homeassistant/loader.py

```diff
@@ -940,22 +990,23 @@ def get_component(self) -> ComponentProtocol:
             )
             raise ImportError(f"Exception importing {self.pkg_path}") from err
 
-        if self.platform_exists("config"):
-            # Setting up a component always checks if the config
-            # platform exists. Since we may be running in the executor
-            # we will use this opportunity to cache the config platform
-            # as well.
-            with suppress(ImportError):
-                self.get_platform("config")
-
-        if self.config_flow:
-            # If there is a config flow, we will cache it as well since
-            # config entry setup always has to load the flow to get the
-            # major/minor version for migrations. Since we may be running
-            # in the executor we will use this opportunity to cache the
-            # config_flow as well.
-            with suppress(ImportError):
-                self.get_platform("config_flow")
+        if preload_platforms:
+            for platform_name in self.platforms_exists(self._preload_platforms):
```

Comment:
Too many things called `preload_platforms` 😅  Can one be called platforms_to_preload maybe?